docs/0.4/concepts/splinter_registry.md

```diff
@@ -0,0 +1,227 @@
+
+A Splinter registry (or just "registry") is a list of nodes that can be browsed
+(and sometimes directly managed) by an administrator. Data for each of the nodes
+in the registry include:
+
+* Display names
+* Unique identities
+* Lists of network endpoints
+* Lists of public keys associated with the nodes
+* Arbitrary metadata
+
+Registries may provide node data from a variety of sources, including local and
+remote files, databases, and even other registries, as is the case with the
+unified registry.
+
+## Unified Registry
+
+<img alt="Splinter registry admin diagram" src="/assets/registry_admin_diagram.svg">
+
+The Splinter daemon (splinterd) uses a **unified registry**, which aggregates
+data from multiple "source" or "upstream" registries. There are two types of
+source registries:
+
+* The **internal registry** is a local, modifiable list of nodes that belongs
+  solely to the Splinter daemon. A unified registry has only one internal
+  registry. All modifications to the unified registry, like adding or updating a
+  node definition, are stored in the internal registry.
+
+* **External registries** are, as the name implies, managed externally. They are
+  read-only to the unified registry, and they cannot be modified using the
+  registry APIs. A unified registry may have zero or more external registries.
+
+### Data Merging and Precedence
+
+The unified registry merges data from its internal and external sources,
+presenting a single, aggregated view of all the data.
+
+If a node with the same ID exists in multiple upstream registries, the
+definition of the node from the highest-precedence registry is used. The
+internal registry has the highest precedence; the precedence of the external
+registries is determined by the order in which they're specified (see
+[Configuring the splinterd Registry](#configuring-the-splinterd-registry) to
+learn how external registries are configured).
+
+Node metadata (arbitrary key/value pairs associated with a node)
+is merged from all sources; if a node with the same ID exists in multiple source
+registries, the metadata for that node is merged from all of them. If the same
+metadata key is set for a node in multiple registries, precedence is determined
+in the same way as node definitions.
+
+### Recommended Patterns
+
+External registries are the recommended way to manage lists of known Splinter
+nodes because they can be shared in a consistent way across a Splinter network.
+The internal registry is only available on the local Splinter node, so any data
+added, modified, or removed from it will not be reflected on other nodes. This
+can be useful in some cases, but generally it's desirable for node data to be
+consistent across the network.
+
+While most node data is static, it is sometimes necessary to modify the
+properties of a node that's defined in an external registry. This can be done
+using the internal registry, but it's usually better to modify the external
+source registry directly.
```

Comment:
For term consistency, drop "source" ("... the external registry directly").

Reply:
Updated